base_layer/core/src/transactions/transaction_components/encrypted_value.rs

```diff
@@ -98,6 +100,7 @@ impl EncryptedValue {
         Ok(EncryptedValue(data))
     }
 
+    /// Authenticate and decrypt the value
```

Comment:
Recommend adding a comment that the design (like basically all existing AEADs) is not key committing, so the caller must not rely on successful decryption to assert that the expected key was used.